### ISSUE TITLE
bytes.vim breaks SpaceVim for Mac

### ISSUE THREAD

**Issue by bravecorvus** [wsdjeg/SpaceVim#3189]
## Expected behavior, english is required
normal startup

## The reproduce ways from Vim starting (Required!)
`nvim`

## Debug info

```
  1 SpaceVim layers:
  2
  3 - VersionControl:       not loaded          https://spacevim.org/layers/VersionControl/
  4 + autocomplete:             loaded          https://spacevim.org/layers/autocomplete/
  5 - chat:                 not loaded          https://spacevim.org/layers/chat/
  6 + checkers:                 loaded          https://spacevim.org/layers/checkers/
  7 - chinese:              not loaded          https://spacevim.org/layers/chinese/
  8 - colorscheme:          not loaded          https://spacevim.org/layers/colorscheme/
  9 + core#banner:              loaded          https://spacevim.org/layers/core/banner/
 10 + core#statusline:          loaded          https://spacevim.org/layers/core/statusline/
 11 + core#tabline:             loaded          https://spacevim.org/layers/core/tabline/
 12 + core:                     loaded          https://spacevim.org/layers/core/
 13 - cscope:               not loaded          https://spacevim.org/layers/cscope/
 14 - ctrlp:                not loaded          https://spacevim.org/layers/ctrlp/
 15 - ctrlspace:            not loaded          https://spacevim.org/layers/ctrlspace/
 16 - debug:                not loaded          https://spacevim.org/layers/debug/
 17 - denite:               not loaded          https://spacevim.org/layers/denite/
 18 + edit:                     loaded          https://spacevim.org/layers/edit/
 19 - exprfold:             not loaded          no exists
 20 - floobits:             not loaded          https://spacevim.org/layers/floobits/
 21 + format:                   loaded          https://spacevim.org/layers/format/
 22 - fzf:                  not loaded          https://spacevim.org/layers/fzf/
 23 - games:                not loaded          no exists
 24 - git:                  not loaded          https://spacevim.org/layers/git/
 25 - github:               not loaded          https://spacevim.org/layers/github/
 26 - gtags:                not loaded          https://spacevim.org/layers/gtags/
 27 - incsearch:            not loaded          no exists
 28 - indentmove:           not loaded          no exists
 29 - japanese:             not loaded          https://spacevim.org/layers/japanese/
 30 + lang#WebAssembly:         loaded          https://spacevim.org/layers/lang/WebAssembly/
 31 - lang#agda:            not loaded          https://spacevim.org/layers/lang/agda/
 32 - lang#asciidoc:        not loaded          https://spacevim.org/layers/lang/asciidoc/
 33 - lang#assembly:        not loaded          https://spacevim.org/layers/lang/assembly/
 34 - lang#autohotkey:      not loaded          https://spacevim.org/layers/lang/autohotkey/
 35 - lang#batch:           not loaded          https://spacevim.org/layers/lang/batch/
 36 + lang#c:                   loaded          https://spacevim.org/layers/lang/c/
 37 - lang#chapel:          not loaded          https://spacevim.org/layers/lang/chapel/
 38 - lang#clojure:         not loaded          https://spacevim.org/layers/lang/clojure/
 39 - lang#coffeescript:    not loaded          https://spacevim.org/layers/lang/coffeescript/
 40 - lang#crystal:         not loaded          https://spacevim.org/layers/lang/crystal/
 41 - lang#csharp:          not loaded          https://spacevim.org/layers/lang/csharp/
 42 - lang#d:               not loaded          https://spacevim.org/layers/lang/d/
 43 - lang#dart:            not loaded          https://spacevim.org/layers/lang/dart/
 44 + lang#dockerfile:          loaded          https://spacevim.org/layers/lang/dockerfile/
 45 - lang#elixir:          not loaded          https://spacevim.org/layers/lang/elixir/
 46 - lang#elm:             not loaded          https://spacevim.org/layers/lang/elm/
 47 - lang#erlang:          not loaded          https://spacevim.org/layers/lang/erlang/
 48 - lang#extra:           not loaded          https://spacevim.org/layers/lang/extra/
 49 - lang#foxpro:          not loaded          https://spacevim.org/layers/lang/foxpro/
 50 - lang#fsharp:          not loaded          https://spacevim.org/layers/lang/fsharp/
 51 + lang#go:                  loaded          https://spacevim.org/layers/lang/go/
 52 - lang#goby:            not loaded          https://spacevim.org/layers/lang/goby/
 53 - lang#gosu:            not loaded          https://spacevim.org/layers/lang/gosu/
 54 - lang#graphql:         not loaded          https://spacevim.org/layers/lang/graphql/
 55 - lang#groovy:          not loaded          https://spacevim.org/layers/lang/groovy/
 56 - lang#hack:            not loaded          https://spacevim.org/layers/lang/hack/
 57 - lang#haskell:         not loaded          https://spacevim.org/layers/lang/haskell/
 58 + lang#html:                loaded          https://spacevim.org/layers/lang/html/
 59 - lang#hy:              not loaded          https://spacevim.org/layers/lang/hy/
 60 - lang#idris:           not loaded          https://spacevim.org/layers/lang/idris/
 61 - lang#io:              not loaded          https://spacevim.org/layers/lang/io/
 62 - lang#ipynb:           not loaded          https://spacevim.org/layers/lang/ipynb/
 63 - lang#j:               not loaded          https://spacevim.org/layers/lang/j/
 64 + lang#java:                loaded          https://spacevim.org/layers/lang/java/
 65 + lang#javascript:          loaded          https://spacevim.org/layers/lang/javascript/
 66 - lang#json:            not loaded          no exists
 67 - lang#julia:           not loaded          https://spacevim.org/layers/lang/julia/
 68 - lang#kotlin:          not loaded          https://spacevim.org/layers/lang/kotlin/
 69 - lang#latex:           not loaded          https://spacevim.org/layers/lang/latex/
 70 - lang#lisp:            not loaded          https://spacevim.org/layers/lang/lisp/
 71 - lang#livescript:      not loaded          https://spacevim.org/layers/lang/livescript/
 72 - lang#lua:             not loaded          https://spacevim.org/layers/lang/lua/
 73 - lang#markdown:        not loaded          https://spacevim.org/layers/lang/markdown/
 74 - lang#matlab:          not loaded          https://spacevim.org/layers/lang/matlab/
 75 - lang#nim:             not loaded          https://spacevim.org/layers/lang/nim/
 76 - lang#nix:             not loaded          https://spacevim.org/layers/lang/nix/
 77 - lang#ocaml:           not loaded          https://spacevim.org/layers/lang/ocaml/
 78 - lang#pact:            not loaded          https://spacevim.org/layers/lang/pact/
 79 - lang#pascal:          not loaded          no exists
 80 - lang#perl:            not loaded          https://spacevim.org/layers/lang/perl/
 81 - lang#php:             not loaded          https://spacevim.org/layers/lang/php/
 82 - lang#plantuml:        not loaded          https://spacevim.org/layers/lang/plantuml/
 83 - lang#pony:            not loaded          https://spacevim.org/layers/lang/pony/
 84 - lang#powershell:      not loaded          https://spacevim.org/layers/lang/powershell/
 85 - lang#processing:      not loaded          https://spacevim.org/layers/lang/processing/
 86 - lang#prolog:          not loaded          https://spacevim.org/layers/lang/prolog/
 87 - lang#puppet:          not loaded          https://spacevim.org/layers/lang/puppet/
 88 - lang#purescript:      not loaded          https://spacevim.org/layers/lang/purescript/
 89 + lang#python:              loaded          https://spacevim.org/layers/lang/python/
 90 - lang#r:               not loaded          https://spacevim.org/layers/lang/r/
 91 - lang#racket:          not loaded          https://spacevim.org/layers/lang/racket/
 92 - lang#red:             not loaded          https://spacevim.org/layers/lang/red/
 93 + lang#ruby:                loaded          https://spacevim.org/layers/lang/ruby/
 94 + lang#rust:                loaded          https://spacevim.org/layers/lang/rust/
 95 + lang#scala:               loaded          https://spacevim.org/layers/lang/scala/
 96 - lang#scheme:          not loaded          https://spacevim.org/layers/lang/scheme/
 97 - lang#sh:              not loaded          https://spacevim.org/layers/lang/sh/
 98 - lang#slim:            not loaded          no exists
 99 - lang#solidity:        not loaded          no exists
100 - lang#sql:             not loaded          no exists
101 - lang#swift:           not loaded          https://spacevim.org/layers/lang/swift/
102 - lang#swig:            not loaded          no exists
103 - lang#tcl:             not loaded          https://spacevim.org/layers/lang/tcl/
104 - lang#toml:            not loaded          no exists
105 - lang#typescript:      not loaded          https://spacevim.org/layers/lang/typescript/
106 - lang#v:               not loaded          https://spacevim.org/layers/lang/v/
107 - lang#vim:             not loaded          https://spacevim.org/layers/lang/vim/
108 - lang#vue:             not loaded          https://spacevim.org/layers/lang/vue/
109 - lang#wolfram:         not loaded          https://spacevim.org/layers/lang/wolfram/
110 - lang#xml:             not loaded          no exists
111 - leaderf:              not loaded          https://spacevim.org/layers/leaderf/
112 + lsp:                      loaded          no exists
113 - mail:                 not loaded          no exists
114 - operator:             not loaded          no exists
115 - org:                  not loaded          no exists
116 + shell:                    loaded          https://spacevim.org/layers/shell/
117 - sudo:                 not loaded          https://spacevim.org/layers/sudo/
118 - test:                 not loaded          https://spacevim.org/layers/test/
119 - tmux:                 not loaded          https://spacevim.org/layers/tmux/
120 + tools#dash:               loaded          https://spacevim.org/layers/tools/dash/
121 - tools#mpv:            not loaded          no exists
122 - tools#screensaver:    not loaded          no exists
123 - tools#zeal:           not loaded          https://spacevim.org/layers/tools/zeal/
124 - tools:                not loaded          https://spacevim.org/layers/tools/
125 + ui:                       loaded          https://spacevim.org/layers/ui/
126 - unite:                not loaded          https://spacevim.org/layers/unite/
127 - vim:                  not loaded          no exists
```

Please press <kbd>SPC h I</kbd>, debug info will be put into clipboard, then paste all content below.

## Screenshots
![img](https://66.media.tumblr.com/85cd6463a2a00b14457c254d347e0448/3080819b6220bf73-55/s1280x1920/08e56b9e764c741346961788111c7514764590dc.png)
If you have any screenshots for this issue, please upload here. BTW you can use https://asciinema.org/ for recording video in terminal.

## More Info:
macOS 10.14.6
nvim: v0.4.2
SpaceVim: Master branch committed yesterday.

**Comment by bravecorvus** [comment]
Also, maybe broken for Linux/Windows (if this even works on Windows)

**Comment by tokelove** [comment]
I can confirm this is also broken on two linux machines, one on Ubuntu 18.04, the other on Ubuntu 19.10:

```
 vim 
Error detected while processing /home/aaron/.SpaceVim/autoload/SpaceVim/api/iconv/bytes.vim:
line    1:
E15: Invalid expression: expand('<sfile>:p:r:gs?[\\/]?#?:s?^.*#autoload#??:s?$?#?')^M
Error detected while processing function SpaceVim#end[94]..SpaceVim#plugins#load[4]..<SNR>51_load_plugins:
line   16:
E170: Missing :endfor
Error detected while processing function SpaceVim#end[94]..SpaceVim#plugins#load:
line    4:
E171: Missing :endif
Press ENTER or type command to continue
```

**Comment by thawk** [comment]
Hi, after some searching, I think this issue is caused by invalid line endings.  Some file, including bytes.vim, has a ``fileformat``of ``windows`` in my Macbook, while the others all have ``fileformat`` of ``unix``.
I have create a PR #3190 to fix it.

**Comment by hlmtre** [comment]
+1 for being broken on Linux. Arch, kernel 5.3.7. Latest neovim release (v0.4.2) and SpaceVim from master.

**Comment by wsdjeg** [comment]
fixed by #3190

**Comment by bravecorvus** [comment]
Thanks @wsdjeg @thawk
